core/src/test/java/org/elasticsearch/search/aggregations/bucket/FilterIT.java

```diff
@@ -118,10 +119,12 @@ public void testEmptyFilterDeclarations() throws Exception {
     }
 
     public void testWithSubAggregation() throws Exception {
+        Factory avgAgg = avg("avg_value");
+        avgAgg.field("value");
```

Comment:
Yep, agreed as this is actually getting quite annoying having to factor out the aggregations in all the test and will make pretty messy tests. It will require all the AggregatoryFactory subclasses to be generic though so they can return their actual type from the inherited methods rather than AggregatorFactory. Because of this the change will affect all AggregatorFactory and PipelineAggregatorFactory implementations. Should I do this in a separate PR to avoid bloating this one?